packages/suite/src/actions/suite/analyticsActions.ts

```diff
@@ -4,559 +4,88 @@
-        }
-    };
+    dispatch({
+        type: ANALYTICS.DISABLE,
+    });
+};
 
 /**
  * Init analytics, should be always run on application start (see suiteMiddleware). It:
  * - sets common analytics variables based on what was loaded from storage
  * - set sentry user id
- * - registers event listeners for reporting events from electron
- * @param loadedState - analytics state loaded from storage
+ * @param state - tracking state loaded from storage
  */
-export const init = (loadedState: State) => (dispatch: Dispatch, getState: GetState) => {
+export const init = (state: AnalyticsState) => (dispatch: Dispatch) => {
     // if instanceId does not exist yet (was not loaded from storage), create a new one
-    const instanceId = loadedState.instanceId || getAnalyticsRandomId();
-    // always create new session id
-    const sessionId = getAnalyticsRandomId();
+    const instanceId = state.instanceId || getTrackingRandomId();
+    const sessionId = getTrackingRandomId();
+
     // if user made choice, keep it, otherwise set it to true by default just to prefill the confirmation toggle
-    const confirmed = !!loadedState.confirmed;
-    const enabled = confirmed ? !!loadedState.enabled : true;
-    // set application state
+    const isConfirmed = !!state.confirmed;
+    const isEnabled = isConfirmed ? !!state.enabled : true;
+
+    // allow tracking only if user already confirmed data collection
+    const userAllowedTracking = isConfirmed && isEnabled;
+
+    analytics.init(userAllowedTracking, {
+        instanceId,
+        sessionId,
+        environment: getEnvironment(),
+        commitId: process.env.COMMITHASH || '',
+        isDev,
+        callbacks: {
+            onEnable: () => dispatch(enable()),
+            onDisable: () => dispatch(disable()),
+        },
+    });
+
+    allowSentryReport(userAllowedTracking);
+    setSentryUser(instanceId);
+
     dispatch({
         type: ANALYTICS.INIT,
         payload: {
             instanceId,
             sessionId,
-            sessionStart: Date.now(),
-            enabled,
-            confirmed,
+            enabled: isEnabled,
+            confirmed: isConfirmed,
         },
     });
-    // allow sentry reporting only if user already confirmed enabling of analytics
-    const userAllowedAnalytics = confirmed && enabled;
-    allowSentryReport(userAllowedAnalytics);
-    // set sentry user id same as session id for analytics
-    setSentryUser(instanceId);
-    // if analytics was initiated as enabled, continue with setting up side effects
-    if (!getState().analytics.enabled) return;
-    // register event listeners to report end of session
-    setOnBeforeUnloadListener(() => {
-        dispatch(
-            report({
-                type: 'session-end',
-                payload: {
-                    start: getState().analytics.sessionStart!,
-                    end: Date.now(),
-                },
-            }),
-        );
-    });
```

Comment:
is this moved somewhere?

Reply:
Good eyes. First of all, I moved it directly to `@trezor/analytics` package, but then I removed it as it would be removed by https://github.com/trezor/trezor-suite/issues/5334 anyway. I have it in mind and I will add it to the changelog as part of the #5334.

Reply:
would be nice to mention at least in the description of this PR/issue or in the changelog right away...